server/rest/workspace/read_test.go

```diff
@@ -0,0 +1,104 @@
+	var (
+		ctrl    *gomock.Controller
+		w       *restworkspacesv1alpha1.Workspace
+		request *http.Request
+		fake    *mocks.MockFakeResponseWriter
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		w = &restworkspacesv1alpha1.Workspace{}
+		w.Name = "foo"
+		w.Namespace = "bar"
+
+		request = buildGetRequest(w)
+		fake = mocks.NewMockFakeResponseWriter(ctrl)
+	})
+
+	AfterEach(func() { ctrl.Finish() })
+
+	DescribeTable("workspace GET handler: read",
+		func(
+			mapperFunc workspace.ReadWorkspaceMapperFunc,
+			readHandler workspace.ReadWorkspaceQueryHandlerFunc,
+			marshaler marshal.MarshalerProvider,
+			responseFunc func() http.ResponseWriter,
+		) {
+			response := responseFunc()
+			handler := workspace.NewReadWorkspaceHandler(mapperFunc, readHandler, marshaler)
+			handler.ServeHTTP(response, request)
+		},
+		Entry("failure in marshal provider", workspace.MapReadWorkspaceHttp, nopReadHandler, errorMarshalProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusBadRequest)
+			return fake
+		}),
+		Entry("failure in read handler", workspace.MapReadWorkspaceHttp, badReadHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure marshaling response", workspace.MapReadWorkspaceHttp, nopReadHandler, badMarshalProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure to write response", workspace.MapReadWorkspaceHttp, nopReadHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().Header().Return(http.Header{})
+			fake.EXPECT().Write(gomock.Any()).Return(0, fmt.Errorf("failed to write response body"))
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure to write response", workspace.MapReadWorkspaceHttp, nopReadHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().Header().Return(http.Header{})
+			fake.EXPECT().Write(gomock.Any()).DoAndReturn(func(a any) (int, error) {
+				slice, ok := a.([]byte)
+				Expect(ok).To(BeTrue())
+				return len(slice), nil
+			})
+			return fake
+		}),
+	)
+})
+
+func badReadHandler(ctx context.Context, cmd coreworkspace.ReadWorkspaceQuery) (*coreworkspace.ReadWorkspaceResponse, error) {
+	return nil, fmt.Errorf("bad create handler")
+}
+
+func nopReadHandler(_ctx context.Context, cmd coreworkspace.ReadWorkspaceQuery) (*coreworkspace.ReadWorkspaceResponse, error) {
+	return &coreworkspace.ReadWorkspaceResponse{
+		Workspace: nil,
+	}, nil
+}
+
+func buildGetRequest(workspace *restworkspacesv1alpha1.Workspace) *http.Request {
+	byteSlice, err := marshal.DefaultMarshal.Marshal(workspace)
+	Expect(err).NotTo(HaveOccurred())
+
+	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces", workspace.GetNamespace())
```

Comment:
This isn't the url for get requests, they need to have the name of the workspace:
```suggestion
	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces/%s", workspace.GetNamespace(), workspace.GetName())
```